core/src/main/java/com/amplifyframework/predictions/models/Sentiment.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.predictions.models;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Class that holds the sentiment detection results for a
+ * string of text for the predictions category.
+ */
+public final class Sentiment {
+    private final SentimentType predominantSentiment;
+    private final Map<SentimentType, Float> sentimentScores;
+
+    private Sentiment(
+            @NonNull SentimentType predominantSentiment,
+            @Nullable Map<SentimentType, Float> sentimentScores
+    ) {
+        this.predominantSentiment = predominantSentiment;
+        this.sentimentScores = sentimentScores;
+    }
+
+    /**
+     * Gets the detected predominant sentiment of the text.
+     * @return the predominant sentiment type
+     */
+    @NonNull
+    public SentimentType getPredominantSentiment() {
+        return predominantSentiment;
+    }
+
+    /**
+     * Gets a map of associated sentiments and their confidence
+     * scores.
+     * @return the map of associated sentiments and scores
+     */
+    @Nullable
+    public Map<SentimentType, Float> getSentimentScores() {
+        return sentimentScores;
```

Comment:
Should arrange for this to be an `Immutable.of(...)`, probably by doing making it immutable in the `Builder`'s  `build()`.